ghpages.mk

```diff
@@ -84,6 +99,22 @@ ifeq (true,$(PUSH_GHPAGES))
 ifneq (,$(TARGET_DIR))
 	mkdir -p $(GHPAGES_TMP)/$(TARGET_DIR)
 endif
+
+	@EXCESS_FILES="$(addprefix $(TARGET_DIR),$(filter-out $^,$(notdir $(foreach ext,*.html *.txt,$(wildcard $(GHPAGES_TMP)/$(TARGET_DIR)/$(ext))))))"; \
+	if [ -n "$$EXCESS_FILES" ]; \
+		then git -C $(GHPAGES_TMP) rm -f --ignore-unmatch -- $$EXCESS_FILES; \
+	fi
```

Comment:
I wonder if this can't be simplified greatly by making this just:
```
git -C $(GHPAGES_TMP) rm -fq --ignore-unmatch -- $(GHPAGES_TMP)/$(TARGET_DIR)/*.txt $(GHPAGES_TMP)/$(TARGET_DIR)/*.html
```
The files that are being added here will be added back.  I just checked and `git rm` followed by `git add` works fine.